packages/utilities/exposure-scanning/scan/index.js

```diff
@@ -0,0 +1,77 @@
+const replacementStr = '*';
+
+export const scanText = (text, regex) => {
+  let foundTextMatch = false;
+  const scannedText = text.replace(regex, replacementStr);
+
+  if (scannedText !== text) {
+    foundTextMatch = true;
+  }
+
+  return { body: scannedText, foundTextMatch };
+};
+
+export const scanComments = (comments, regex) => {
+  let foundCommentMatch = false;
+  const scannedComments = [];
+
+  comments.forEach(({ id, body }) => {
+    if (regex.test(body)) {
+      foundCommentMatch = true;
+      scannedComments.push({
+        id,
+        body: body.replace(regex, replacementStr),
+      });
+    }
+  });
+
+  return { comments: scannedComments, foundCommentMatch };
+};
+
+export const scanPr = (pr, regex) => {
+  let scannedPr = pr;
```

Comment:
this variable will always be mutated so it doesn't make sense to not just return inside the try like this

```js
export const scanPr = (pr, regex) => {
  try {
    const { body, foundTextMatch } = scanText(pr.body, regex);
    const { comments, foundCommentMatch } = scanComments(pr.comments, regex);
    const {
      comments: reviewComments,
      foundCommentMatch: foundReviewCommentsMatch,
    } = scanComments(pr.reviewComments, regex);

    return {
      body,
      comments,
      reviewComments,
      foundMatch:
        foundTextMatch || foundCommentMatch || foundReviewCommentsMatch,
    };
  } catch (error) {
    throw new Error('Encountered an error when scanning.');
  }
};
```

Reply:
Thanks @jroebu14! This has been a really insightful review 🙌 Will make those changes now 🙂